lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -374,7 +374,11 @@ sub get_test_results {
 sub get_test_history {
     my ( $self, $p ) = @_;
 
-    my $results = $self->{db}->get_test_history( $p );
+    my $results;
+    
+    return $results unless ($p->{frontend_params} && $p->{frontend_params}{domain});
```

Comment:
It would make more sense to return an error here. Would that break the current GUI implementation?

Reply:
I know that the current GUI is dependent on an empty query being replied to in this way, so I think it is OK, but I think this should be documented in the code that solution is of that reason, that it is deprecated to be dependent on that behavior and that the plan is to do as @mattias-p suggests.